src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -422,7 +422,7 @@ async Task<InProgressState> CollapseInProgressStateAsync(InProgressState initial
                             // Also transform the compilation that has generated files; we won't do that though if the transformation either would cause problems with
                             // the generated documents, or if don't have any source generators in the first place.
                             if (translationAction.CanUpdateCompilationWithStaleGeneratedTreesIfGeneratorsGiveSameOutput &&
-                                translationAction.OldProjectState.SourceGenerators.Any())
+                                GetSourceGenerators(translationAction.OldProjectState).Any())
```

Comment:
Note: this needs to not instantiate generators, but instead call into OOP to determine this info.

Reply:
Considering that the SDK delivers source generators that are enabled by default, it might make sense to just assume the `.Any()` is true.